File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.c
#include <stdbool.h>

int longestSubsequence(int* nums, int numsSize) {
    int xor_val = 0; // renamed from 'xor' to avoid C++ keyword conflicts in some compilers
    bool hasNonZero = false;

    for (int i = 0; i < numsSize; i++) {
        int num = nums[i];
        xor_val ^= num;

        if (num != 0) {
            hasNonZero = true;
        }
    }

    if (xor_val != 0) {
        return numsSize;
    }

    return hasNonZero ? numsSize - 1 : 0;
}

File: 3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestSubsequence = function(nums) {
    let xor = 0;
    let hasNonZero = false;

    for (let i = 0; i < nums.length; i++) {
        const num = nums[i];
        xor ^= num;

        if (num !== 0) {
            hasNonZero = true;
        }
    }

    if (xor !== 0) {
        return nums.length;
    }

    return hasNonZero ? nums.length - 1 : 0;
};
